pkg/sdk/go/errors.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package sdk
+
+import (
+	"encoding/json"
+
+	"github.com/mainflux/mainflux/pkg/errors"
+)
+
+var errEncodeError = errors.NewSDKError("failed to encode response error")
+
+var (
+	// ErrFailedFetch indicates that fetching of entity data failed.
+	ErrFailedFetch = errors.NewSDKError("failed to fetch entity")
+
+	// ErrFailedConnect indicates that connecting thing to channel failed.
+	ErrFailedConnect = errors.NewSDKError("failed to connect thing to channel")
+
+	// ErrFailedDisconnect indicates that disconnecting thing from a channel failed.
+	ErrFailedDisconnect = errors.NewSDKError("failed to disconnect thing from channel")
+
+	// ErrInvalidContentType indicates that non-existent message content type was passed.
+	ErrInvalidContentType = errors.NewSDKError("Unknown Content Type")
+
+	// ErrFailedWhitelist failed to whitelist configs
+	ErrFailedWhitelist = errors.NewSDKError("failed to whitelist")
+
+	// ErrCerts indicates error fetching certificates.
+	ErrCerts = errors.NewSDKError("failed to fetch certs data")
+
+	// ErrCertsRemove indicates failure while cleaning up from the Certs service.
+	ErrCertsRemove = errors.NewSDKError("failed to remove certificate")
+
+	// ErrFailedCertUpdate failed to update certs in bootstrap config
+	ErrFailedCertUpdate = errors.NewSDKError("failed to update certs in bootstrap config")
+
+	// ErrMemberAdd failed to add member to a group.
+	ErrMemberAdd = errors.NewSDKError("failed to add member to group")
+)
+
+func encodeError(body []byte, status int) errors.SDKError {
```

Comment:
This is similar to `pkg/errors/sdk_errors.go` method `CheckErr`. Do we need both?

Reply:
encodeError() simply joins the http statuscode to the error, and returns a new composite SDKError.
In Checkerror(), it checks for a range of expected statuscodes, if any one of matches with the resp.statuscode, the function doesn't return any error, else, it checks further, and returns a composite error.

So, both are used for different reasons, and hence why I had to keep both.
(The tests and other functions get very complicated and difficult-to-read, when I use just one function, for both these usecases)

Reply:
I am not sure this is the case. Please check the comment I left on `ViewBootstrap` for reference.